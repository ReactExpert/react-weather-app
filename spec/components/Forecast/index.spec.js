import React from 'react';
import Forecast from '../../../src/components/Forecast';
import { enzymeConfig, mount } from '../../enzymeConfig';

enzymeConfig();

const ForecastComponent = (<Forecast id='11' title='some title' icon='wi-cloudy' animationDelay={1} > <h1>Children</h1> </Forecast>);

describe('Forecast', () => {
  it('matches snapshot', () => {
    const component = mount(ForecastComponent);

    expect(component).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls animate', (done) => {
      const component = mount(ForecastComponent);
      const instance = component.instance();

      instance.animate = jest.fn();

      setTimeout(() => {
        expect(instance.animate).toBeCalled();
        done();
      }, 200);
    });
  });

  describe('.icon', () => {
    it('matches source icon path wi-cloudy.svg', () => {
      const component = mount(ForecastComponent);

      component.debug();

      expect(component.find('[src="wi-cloudy.svg"]').length).toEqual(1);
    });
  });
});
