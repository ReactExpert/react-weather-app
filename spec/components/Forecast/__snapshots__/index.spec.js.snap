// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast matches snapshot 1`] = `
<Forecast
  animationDelay={1}
  icon="wi-cloudy"
  id="11"
  title="some title"
>
  <div
    className="forecast"
    data-id="11"
  >
    <span
      className="forecast__title"
    >
      some title
    </span>
    <img
      alt="icon"
      className="forecast__icon"
      src="wi-cloudy.svg"
    />
     
    <h1>
      Children
    </h1>
     
  </div>
</Forecast>
`;
